docs/CPS/CPS-XXXX/README.md

```diff
@@ -0,0 +1,101 @@
+---
+CPS:
+Title: Sub-Optimal Provider to Wallet API
+Status: Open
+Category: Wallets
+Authors:
+    - Rhys Bartels-Waller <rhys.bartelswaller@iohk.io>
+    - William Wolff <william.wolff@iohk.io>
+    - Martynas Kazlauskas <martynas.kaslauskas@iohk.io>
+    - Daniele Ricci <daniele.ricci@iohk.io>
+    - Sefa Irken <sefa.irken@iohk.io>
+
+
+Proposed Solutions: []
+Discussions:
+    - https://github.com/cardano-foundation/CIPs/tree/master/CPS-0011
+Created: 2024-07-09
+License: CC-BY-4.0
+---
+
+## Abstract
+The event-based nature of a Blockchain is well-suited for wallets to derive their internal state from genesis, however, most data services only expose a request/response interface and/or take on the responsibility of decoding the native binary data into JSON, enforcing a strong client/server contract with the current protocol transaction format. Wallets must use long polling to detect changes, on an interval that balances latency with network and service demand, then make numerous calls against various endpoints that have the potential to change on each protocol upgrade. Furthermore, the lack of an API standard inhibits development and adoption of a standalone desktop service for wallets to offer a full node experience for their users, which is best achieved via a universal API running either on a remote server, in a multi-tenant mode, or locally running in single-tenant mode.
+
+## Problem
+> [!NOTE]
+>
+> Fundamental wallet functionality is defined in this document as being able to construct and submit any transaction supported by the latest Cardano protocols, requiring:
+> - **UTxO set** owned by addresses with credentials controlled by the wallet, that are either derived within the 1852 purposed, [BIP32-ed25519](https://input-output-hk.github.io/adrestia/static/Ed25519_BIP.pdf) accounts or the product of hashing a script.
+> - **Active stake pools**: to verify a nominated Stake Pools ID is valid and active at the time of delegation.
```

Comment:
I'm not sure this is required in base API :thinking: Without metadata/metrics, wallet can't really help the user to choose a stake pool. Yes this could validate it's 'active', but users would have to use an external source anyway. 
```suggestion
```

Reply:
The idea is to verify a delegation is legitimate, rather than helping the user choose the pool. Now I think about it, we would also need to include DReps if this is the thinking.

Reply:
I understand, but I don't think it's something that wallets definitely have to do. If node allows delegating to an unexisting/inactive stake pool (I believe it does but not 100% sure), then some wallets might want to allow that as well. A likely use case is for registering a stake pool and delegating to it before it become active.

Reply:
Ah, good point. I will remove the "active" aspect of it, leaving the responsibility to just a registered entity

Reply:
https://github.com/will-break-it/wallet-architecture/pull/3/commits/2292d1203fa5d9bc56636a43b97de76743d5f112